### PR TITLE
Misc linter-related fixes and enhancements

### PR DESCRIPTION
**Pull request by lukpueh**
**Fixes issue #**:
None

**Description of the changes being introduced by the pull request**:
This PR
- enables  trailing whitespace check in `pylint` and fixes trailing whitespace  (fb7e46c),
- updates the usage of `bandit` in `tox` (5518bf3)
- removes the obsolete ci requirement pytest (1cb0ec3)
- enables `pylint` for the tests and fixes thereby unmasked errors (2f12af2 ff.)

Despite the large diff, the PR should be easy to review, as most changes are rather mechanical/repetitive.

I strongly recommend to review commit by commit.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by lukpueh** [comment]
Headsup: 12d2f23745b39b8dcb3c4203c38355223fe2ce13 and c666aeb0bc4ae1014835979bad5119936792f21c duplicate changes that are also introduced in #266's 23cb4d7e62233b5891e2d7c5d5b6bc07fb1d909d. Expect minor conflicts.

File: tests/runtests.py [hunk 1]
```diff
@@ -33,8 +33,8 @@ def check_usable_gpg():
     try:
       subprocess.check_call([gpg, "--version"])
 
-    except OSError as e:
-      pass
```
Comment (SantiagoTorres) [inline tests/runtests.py]: it feels like this change doesn't belong to this commit, does it?

**Review by SantiagoTorres — COMMENTED** [review]
This looks very good! Great job! 

I have an incredibly minor nit that may be an error on my side

File: tests/runtests.py (hunk 1 above)
Comment (lukpueh) [reply tests/runtests.py]: Damned, you're right. That must have slipped in while rebasing. Let me see what happened.

Comment (lukpueh) [reply tests/runtests.py]: Thanks for your good eyes, @SantiagoTorres. This indeed slipped in while rebasing onto (a.o.) 23cb4d7e62233b5891e2d7c5d5b6bc07fb1d909d.

Just fixed it.

**Comment by SantiagoTorres** [comment]
LGTM, thanks!
